pkg/migration/migration.go

```diff
@@ -174,19 +174,31 @@ func (h *Handler) processMigration(
 	// correctly as we go through different phases of migration we can use that instead of this
 	// internal annotation.
 	// Block the component migration until the portworx pod migration is finished.
-	cluster.Annotations[constants.AnnotationPauseComponentMigration] = "true"
-	if err := h.client.Update(context.TODO(), cluster, &client.UpdateOptions{}); err != nil {
+	liveCluster := &corev1.StorageCluster{}
```

Comment:
Can we just use the `cluster` object here so that this latest cluster is passed along to the next steps in the migration? Otherwise the `cluster` object is going to be stale.

Reply:
after calling `processMigration` there's no other place we use the cluster object so I think it's ok. also since we have both migration handler and controller updating the storage cluster, most likely we always have to get the latest version, so it can be stale anytime.

Reply:
major issue to resolve here is conditions are updated by components, so we only overwrite the conditions here (phase should be optional). then underneath in `Driver.UpdateStorageClusterStatus`, px runtime condition will be updated first, then we iterate all conditions to conclude the phase.

Reply:
Got it.
I saw that methods like `portworxNodesRollingUpdate` or `portworxNodesOnDeleteUpdate` were using it, but found out that they don't update the cluster. So it should be fine.